pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -459,38 +533,52 @@ func (c *Cluster) updateMonDeploymentReplica(name string, enabled bool) error {
 func (c *Cluster) failoverMon(name string) error {
 	logger.Infof("Failing over monitor %q", name)
 
+	// remove the failed mon from a local list of the existing mons for finding a stretch zone
+	existingMons := c.clusterInfoToMonConfigWithExclude(name)
+
+	zone, err := c.findAvailableZoneIfStretched(existingMons)
+	if err != nil {
+		return errors.Wrap(err, "failed to find available stretch zone")
+	}
+
+	// Start a new monitor
+	m := c.newMonConfig(c.maxMonID+1, zone)
+	logger.Infof("starting new mon: %+v", m)
+
 	// Scale down the failed mon to allow a new one to start
 	if err := c.updateMonDeploymentReplica(name, false); err != nil {
 		// attempt to continue with the failover even if the bad mon could not be stopped
 		logger.Warningf("failed to stop mon %q for failover. %v", name, err)
 	}
+
+	// If the mon failover is not successful, revert the failover
 	newMonSucceeded := false
+	newMonMightBeInQuorum := false
 	defer func() {
 		if newMonSucceeded {
 			// do nothing if the new mon was started successfully, the deployment will anyway be deleted
 			return
 		}
+		logger.Warningf("failover of mon %q unsuccessful, cleaning up failover mon %q", name, m.DaemonName)
```

Comment:
Do you think this wording might be better?

```suggestion
		logger.Warningf("failover of mon %q unsuccessful, cleaning up replacement mon %q", name, m.DaemonName)
```